run/grpc-ping/ping.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	pb "github.com/GoogleCloudPlatform/golang-samples/run/grpc-ping/pkg/api/v1"
+	"github.com/golang/protobuf/ptypes"
+)
+
+type pingService struct {
+	pb.UnimplementedPingServiceServer
+}
+
+func (s *pingService) Send(ctx context.Context, req *pb.Request) (*pb.Response, error) {
+	log.Print("sending ping response")
+	return &pb.Response{
+		Pong: &pb.Pong{
+			Index:      1,
+			Message:    req.GetMessage(),
+			ReceivedOn: ptypes.TimestampNow(),
+		},
+	}, nil
+}
+
+func (s *pingService) SendUpstream(ctx context.Context, req *pb.Request) (*pb.Response, error) {
+	if Conn == nil {
+		return nil, fmt.Errorf("no upstream connection configured")
+	}
+
+	p := &pb.Request{
+		Message: req.GetMessage() + " (relayed)",
+	}
+	url := "https://" + os.Getenv("GRPC_PING_HOST")
```

Comment:
Is it standard for the host variable not to include the protocol?

Reply:
I've split HOST and PORT into separate variables. In this case the assembled URL should not use the port and is only needed for minting an ID token. Tweaking to clarify that point.